source/common/network/listener_filter_buffer_impl.cc

```diff
@@ -0,0 +1,61 @@
+#include "source/common/network/listener_filter_buffer_impl.h"
+
+#include <string.h>
+
+namespace Envoy {
+namespace Network {
+
+uint64_t ListenerFilterBufferImpl::copyOut(void* buffer, uint64_t length) {
+  auto size = std::min(length, data_size_);
+  if (data_size_ == 0) {
+    return 0;
+  }
+  memcpy(buffer, base_, size); // NOLINT(safe-memcpy)
```

Comment:
Per my other comment let's actually use safe versions of data structures here.

Reply:
got it